Docs: make it clear that MariaDB is also supported as datasource (#71832)
grafana/grafana: docs/sources/datasources/mysql/_index.md

aliases:
  - ../data-sources/mysql/
  - ../features/datasources/mysql/
description: Guide for using MySQL/MariaDB in Grafana
keywords:
  - grafana
  - mariadb
  - mysql
  - guide
labels:
  products:
    - cloud
    - enterprise
    - oss
menuTitle: MySQL/MariaDB

> title: MySQL/MariaDB data source

Reviewer: I agree that we could probably specify support for MariaDB more clearly in the docs (although if, in the future, it moves away from MySQL compatibility our DS might not support it anymore), but I don't think we should append `MariaDB` to all occurrences. I'd see this as more of a sentence somewhere in the docs saying that other MySQL replacements like MariaDB also work.

I think that support for MariaDB is mainly the result of it being a compatible drop-in replacement for MySQL. I haven't tested, but I would assume the same could be said about other similar products like [Percona Server](https://www.percona.com/software/mysql-database/percona-server). The main thing I'm trying to point out here is that there may be other MySQL derivates that the DS can connect to, but we cannot change its name in the docs to cater to all.

Reviewer 2: This is a good point, and one I didn't fully consider. We DO certainly support MariaDB, but only **_incidentally_**, due to the fact that it maintains compatibility with its MySQL's API and commands. However, if that ever changes, we currently DO NOT have on official MariaDB plugin. @fauust I think @mdvictor is correct in believing that simply appending MariaDB to this section of our docs is misleading, and isn't the correct approach. What do you think of instead mentioning MariaDB's current compatibility here (/docs/sources/datasources/mysql/_index.md, line 24) and here (/public/app/plugins/datasource/mysql/README.md, line 3)?

| **Password**                  | Database user's password                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       |

> | **Session Timezone**          | Specify the time zone used in the database session, such as `Europe/Berlin` or `+02:00`. This is necessary, if the timezone of the database (or the host of the database) is set to something other than UTC. Set the value used in the session with `SET time_zone='...'`. If you leave this field empty, then the time zone is not updated. For more information, refer to the [MySQL documentation](https://dev.mysql.com/doc/en/time-zone-support.html).                                                                                                                                                                                                                   |

Reviewer 3: ```suggestion
| **Session Timezone**          | Specifies the time zone used in the database session, such as `Europe/Berlin` or `+02:00`. Required if the timezone of the database (or the host of the database) is set to something other than UTC. Set the value used in the session with `SET time_zone='...'`. If you leave this field empty, then the time zone is not updated. For more information, refer to the [MySQL documentation](https://dev.mysql.com/doc/en/time-zone-support.html).                                                                                                                                                                                                                   |
```

| **Max idle**                  | The maximum number of connections in the idle connection pool, default `100` (Grafana v5.4+).                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  |

> | **Auto (max idle)**           | If set will set the maximum number of idle connections to the number of maximum open connections (Grafana v9.5.1+). Default is `true`.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         |

Reviewer 3: ```suggestion
| **Auto (max idle)**           | Toggle to set the maximum number of idle connections to the number of maximum open connections (available in Grafana v9.5.1+). Default is `true`.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         |
```

> | **Allow cleartext passwords** | Allows using the [cleartext client side plugin](https://dev.mysql.com/doc/en/cleartext-pluggable-authentication.html) if required by an account, such as one defined with the [PAM authentication plugin](https://dev.mysql.com/doc/en/pam-pluggable-authentication.html). <br />**Sending passwords in clear text may be a security problem in some configurations**. To avoid problems if there is any possibility that the password would be intercepted, clients should connect to MySQL Server using a method that protects the password. Possibilities include [TLS / SSL](https://github.com/go-sql-driver/mysql#tls), IPsec, or a private network. Default is `false`. |

Reviewer 3: ```suggestion
| **Allow cleartext passwords** | Allows the use of the [cleartext client side plugin](https://dev.mysql.com/doc/en/cleartext-pluggable-authentication.html) as required by a specific type of account, such as one defined with the [PAM authentication plugin](https://dev.mysql.com/doc/en/pam-pluggable-authentication.html). <br />**Sending passwords in clear text may be a security problem in some configurations**. To avoid password issues, it is recommended that clients connect to MySQL server using a method that protects the password. Possibilities include [TLS / SSL](https://github.com/go-sql-driver/mysql#tls), IPsec, or a private network. Default is `false`. |
```

> | **Max lifetime**              | The maximum amount of time in seconds a connection may be reused, default `14400`/4 hours. This should always be lower than configured [wait_timeout](https://dev.mysql.com/doc/en/server-system-variables.html#sysvar_wait_timeout) in MySQL (Grafana v5.4+).                                                                                                                                                                                                                                                                                                                                                                                                                 |

Reviewer 3: ```suggestion
| **Max lifetime**              | The maximum amount of time in seconds a connection may be reused. This should always be lower than configured [wait_timeout](https://dev.mysql.com/doc/en/server-system-variables.html#sysvar_wait_timeout) in MySQL (Grafana v5.4+).   The default is `14400` or 4 hours.                                                                                                                                                                                                                                                                                                                                                                                                            |
```

### Dataset and Table selection

{{% admonition type="note" %}}

> If your table or database name contains a reserved word or a [not permitted character](https://dev.mysql.com/doc/en/identifiers.html) the editor will put quotes around them. For example a table name like `table-name` will be quoted with backticks `` `table-name` ``.

Reviewer 3: ```suggestion
If your table or database name contains a reserved word or a [prohibited character](https://dev.mysql.com/doc/en/identifiers.html) the editor will put quotes around them. For example a table name like `table-name` will be quoted with backticks `` `table-name` ``.
```

> If your table or database name contains a reserved word or a [not permitted character](https://dev.mysql.com/doc/en/identifiers.html) the editor will put quotes around them. For example a table name like `table-name` will be quoted with backticks `` `table-name` ``.

Reviewer 3: ```suggestion
If your table or database name contains a reserved word or a [prohibited character](https://dev.mysql.com/doc/en/identifiers.html) the editor will put quotes around the name. For example, the name `table-name` will be quoted with backticks  - `` `table-name` ``.
```
